Posh-ACME/Plugins/Infomaniak.ps1

```diff
@@ -0,0 +1,262 @@
+    .PARAMETER ExtraParams
+        This parameter can be ignored and is only used to prevent errors when splatting with more parameters than this function supports.
+    .EXAMPLE
+        Remove-DnsTxt '_acme-challenge.example.com' 'txt-value' -InfomaniakTokenInsecure 'xxxxxxxx'
+        Removes the specified TXT record with the specified value.
+    #>
+}
+
+function Save-DnsTxt {
+    [CmdletBinding()]
+    param(
+        [Parameter(ValueFromRemainingArguments)]
+        $ExtraParams
+    )
+    <#
+    .SYNOPSIS
+        Not required.
+    .DESCRIPTION
+        This provider does not require calling this function to commit changes to DNS records.
+    .PARAMETER ExtraParams
+        This parameter can be ignored and is only used to prevent errors when splatting with more parameters than this function supports.
+    #>
+}
+
+############################
+# Helper Functions
+############################
+
+# API Docs: https://api.infomaniak.com/doc
+
+Function Find-InfomaniakZone {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory, Position=0)]
+        [string]$RecordName,
+        [Parameter(Mandatory, Position=1)]
+        [hashtable]$RestParameters
+    )
+
+    # setup a module variable to cache the record to zone mapping
+    # so it's quicker to find later
+    if (!$script:InfomaniakRecordZones) { $script:InfomaniakRecordZones = @{} }
+
+    # check for the record in the cache
+    if ($script:InfomaniakRecordZones.ContainsKey($RecordName)) {
+        Write-Debug "Result from Cache $($script:InfomaniakRecordZones.$RecordName.name)"
+        return $script:InfomaniakRecordZones.$RecordName
+    }
+
+    # We need to find the closest/deepest
+    # sub-zone that would hold the record rather than just adding it to the apex. So for something
+    # like _acme-challenge.site1.sub1.sub2.example.com, we'd look for zone matches in the following
+    # order:
+    # - site1.sub1.sub2.example.com
+    # - sub1.sub2.example.com
+    # - sub2.example.com
+    # - example.com
+    $zoneTest = $RecordName;
+    while($zoneTest.Contains('.'))
+    {
+        Write-Debug "Checking $zoneTest"
+        try {
+            $response = Invoke-RestMethod -Uri "$apiRoot/1/product?service_name=domain&customer_name=$zoneTest" `
+                @RestParameters @Script:UseBasic -EA Stop
+            $zoneId = $response.data.id
+            if($zoneId)
+            {
+                Write-Debug "Zone $zoneTest found. Zone ID is $zoneId"
+                $script:InfomaniakRecordZones.$RecordName = @{
+                    name = $zoneTest
+                    id = $zoneId
+                }
+                return $script:InfomaniakRecordZones.$RecordName;
+            }
+        } catch { throw }
+        # remove one sub site
+        $zoneTest = $zoneTest.Split('.',2)[1]
+    }
+
+    # remove _acme-challenge. from requested zone
+    $zone = $RecordName.Split('.',2)[1]
+    Write-Debug "Zone $zone does not exist ..."
```

Comment:
Another remnant of removing `_acme-challenge` to cleanup. You might just change the debug message to something like 

```powershell
Write-Debug "Zone for $RecordName not found..."
```

Reply:
Agree. Changed that for coherency.